stumpy/mmotifs.py

```diff
@@ -0,0 +1,186 @@
+        If None, this defaults to
+        `np.nanmax([np.nanmean(D) - 2 * np.nanstd(D), np.nanmin(D)])`
+        (i.e. at least the closest match will be returned).
+
+    cutoffs: numpy.ndarray or float, default None
+        The largest matrix profile value (distance) for each dimension of the
+        multidimensional matrix profile that a multidimenisonal candidate motif is
+        allowed to have.
+        If cutoffs is only one value, these value will be applied to every dimension.
+
+    max_matches: int, default 10
+        The maximum amount of similar matches (nearest neighbors) of a motif
+        representative to be returned.
+        The first match is always the self-match for each motif.
+
+    max_motifs: int, default 1
+        The maximum number of motifs to return
+
+    atol : float, default 1e-8
+        The absolute tolerance parameter. This value will be added to `max_distance`
+        when comparing distances between subsequences.
+
+    normalize : bool, default True
+        When set to `True`, this z-normalizes subsequences prior to computing distances.
+        Otherwise, this function gets re-routed to its complementary non-normalized
+        equivalent set in the `@core.non_normalized` function decorator.
+
+    p : float, default 2.0
+        The p-norm to apply for computing the Minkowski distance. This parameter is
+        ignored when `normalize == False`.
+
+    Returns
+    -------
+    motif_distances: numpy.ndarray
+        The distances corresponding to a set of subsequence matches for each motif.
+
+    motif_indices: numpy.ndarray
+        The indices corresponding to a set of subsequences matches for each motif.
+
+    motif_subspaces: numpy.ndarray
+        A numpy.ndarray consisting of arrays that contain the `k`-dimensional
+        subspace for each motif.
+
+    motif_mdls: numpy.ndarray
+        A numpy.ndarray consisting of arrays that contain the mdl results for
+        finding the dimension of each motif
+    """
+    T = core._preprocess(T)
+
+    m = T.shape[-1] - P.shape[-1] + 1
+
+    if max_motifs < 1:  # pragma: no cover
+        logger.warning(
+            "The maximum number of motifs, `max_motifs`, "
+            "must be greater than or equal to 1"
+        )
+        logger.warning("`max_motifs` has been set to `1`")
+        max_motifs = 1
+
+    T, M_T, Σ_T = core.preprocess(T, m)
+
+    excl_zone = int(np.ceil(m / config.STUMPY_EXCL_ZONE_DENOM))
+
+    if max_matches is None:
+        max_matches = np.inf
+
+    if cutoffs is None:
+        P_copy = P.copy().astype(np.float64)
+        P_copy[np.isinf(P_copy)] = np.nan
+        cutoffs = np.nanmax(
+            [
+                np.nanmean(P_copy - 2.0 * np.nanstd(P_copy, axis=0), axis=1),
+                np.nanmin(P_copy, axis=1),
+            ],
+            axis=0,
+        )
+    elif isinstance(cutoffs, int) or isinstance(cutoffs, float):

```

Comment:
I think we can replace this with:

```
if np.isscalar(cutoffs):
    cutoffs = np.full(P.shape[0], cutoffs)
```

I also replaced `T.shape[0]` with `P.shape[0]` to be consistent with the cutoff section above. Both produce the same result